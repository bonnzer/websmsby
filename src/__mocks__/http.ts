import { METHODS } from '../constants/Request';

export default {
  request: jest.fn((args) => {
    expect(args).toMatchSnapshot();

    return {
      on: jest.fn((e, callback) => {
        callback('test');

        return {
          end: jest.fn(),
          write: jest.fn((body) => {
            if (e.method === METHODS.POST) {
              expect(body).toMatchSnapshot();
            }
          }),
        };
      }),
    };
  }),
};
