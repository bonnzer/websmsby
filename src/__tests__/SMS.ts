import http from 'http';
import { SMS } from '../index';

jest.mock('http');

describe('SMS', () => {
  describe('Params & body presence', () => {
    describe.each([
      [true, ''],
      [false, 'hey'],
    ])(
      `With testMode=%p and devKey=%p`,
      (testMode: boolean, devKey: string) => {
        const instance = new SMS({
          user: 'test',
          apiKey: 'test',
          devKey,
          testMode,
        });
        const testMe = (method: () => void) => async () => {
          try {
            await method();
          } catch (e) {
            expect(e).toBe('test');
          }

          expect(http.request).toHaveBeenCalled();
        };

        test(
          'sendSMS()',
          testMe(
            instance.sendSMS.bind(instance, {
              message: 'test',
              recipients: 'test',
              sender: 'me',
            }),
          ),
        );
        test(
          'sendBulkSMS()',
          testMe(
            instance.sendBulkSMS.bind(instance, [
              {
                message: 'test',
                recipients: 'test',
                sender: 'me',
              },
            ]),
          ),
        );
        test(
          'getStatus()',
          testMe(
            instance.getStatus.bind(instance, {
              messages_id: ['test', 'test1'],
            }),
          ),
        );
        test(
          'getList()',
          testMe(
            instance.getList.bind(instance, {
              messages_id: 'test',
              date_from: '2020-02-01',
              date_to: '2020-02-01',
            }),
          ),
        );
        test(
          'getCost()',
          testMe(
            instance.getCost.bind(instance, {
              recipients: ['test', 'test'],
              message: 'test',
            }),
          ),
        );
        test(
          'getBulkCost()',
          testMe(
            instance.getBulkCost.bind(instance, [
              {
                recipients: ['test', 'test'],
                message: 'test',
              },
            ]),
          ),
        );
      },
    );
  });
});
