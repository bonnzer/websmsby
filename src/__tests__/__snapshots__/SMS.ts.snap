// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SMS Params & body presence With testMode=false and devKey="hey" getBulkCost() 1`] = `
Object {
  "headers": Object {
    "Content-length": 111,
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "POST",
  "path": "/api/msg_cost_bulk",
}
`;

exports[`SMS Params & body presence With testMode=false and devKey="hey" getCost() 1`] = `
Object {
  "headers": Object {
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "GET",
  "path": "/api/msg_cost?devkey=hey&apikey=test&test=false&user=test&recipients=test%2Ctest&message=test",
}
`;

exports[`SMS Params & body presence With testMode=false and devKey="hey" getList() 1`] = `
Object {
  "headers": Object {
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "GET",
  "path": "/api/msg_list?devkey=hey&apikey=test&test=false&user=test&messages_id=test&date_from=2020-02-01&date_to=2020-02-01",
}
`;

exports[`SMS Params & body presence With testMode=false and devKey="hey" getStatus() 1`] = `
Object {
  "headers": Object {
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "GET",
  "path": "/api/msg_status?devkey=hey&apikey=test&test=false&user=test&messages_id=test%2Ctest1",
}
`;

exports[`SMS Params & body presence With testMode=false and devKey="hey" sendBulkSMS() 1`] = `
Object {
  "headers": Object {
    "Content-length": 116,
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "POST",
  "path": "/api/msg_send_bulk",
}
`;

exports[`SMS Params & body presence With testMode=false and devKey="hey" sendSMS() 1`] = `
Object {
  "headers": Object {
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "GET",
  "path": "/api/msg_send?devkey=hey&apikey=test&test=false&user=test&message=test&recipients=test&sender=me",
}
`;

exports[`SMS Params & body presence With testMode=true and devKey="" getBulkCost() 1`] = `
Object {
  "headers": Object {
    "Content-length": 107,
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "POST",
  "path": "/api/msg_cost_bulk",
}
`;

exports[`SMS Params & body presence With testMode=true and devKey="" getCost() 1`] = `
Object {
  "headers": Object {
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "GET",
  "path": "/api/msg_cost?devkey=&apikey=test&test=true&user=test&recipients=test%2Ctest&message=test",
}
`;

exports[`SMS Params & body presence With testMode=true and devKey="" getList() 1`] = `
Object {
  "headers": Object {
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "GET",
  "path": "/api/msg_list?devkey=&apikey=test&test=true&user=test&messages_id=test&date_from=2020-02-01&date_to=2020-02-01",
}
`;

exports[`SMS Params & body presence With testMode=true and devKey="" getStatus() 1`] = `
Object {
  "headers": Object {
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "GET",
  "path": "/api/msg_status?devkey=&apikey=test&test=true&user=test&messages_id=test%2Ctest1",
}
`;

exports[`SMS Params & body presence With testMode=true and devKey="" sendBulkSMS() 1`] = `
Object {
  "headers": Object {
    "Content-length": 112,
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "POST",
  "path": "/api/msg_send_bulk",
}
`;

exports[`SMS Params & body presence With testMode=true and devKey="" sendSMS() 1`] = `
Object {
  "headers": Object {
    "Content-type": "application/json",
  },
  "hostname": "cp.websms.by",
  "method": "GET",
  "path": "/api/msg_send?devkey=&apikey=test&test=true&user=test&message=test&recipients=test&sender=me",
}
`;
